### DISCUSSION TITLE
Getting Error Box on ComfyUI startup

### DISCUSSION THREAD

**Discussion by chnisar515** [comfyanonymous/ComfyUI#5301] [General]
Hello Experts,
Now its 2nd time i am getting this error box and i am trying to fix it with old method but it did not work.
Today i just updated my ComfyUI after that when i try to restart my ComfyUI then on startup its showing me an error box i have attached the file so please guide me how i can fix this issue. by the way my ComfyUI working good but that error just irritating me. and also when my CMD loaded fully it showing me the URL (http://127.0.0.1:8188) to run ComfyUI on browser manually but before this it will automatically launch on my default browser also guide me how i can fix this too.
 
![Screenshot 2024-10-21 020622](https://github.com/user-attachments/assets/801f7a42-b9d5-4e3b-b927-364a7ac75a05)

below are the log of my cmd

`C:\ComfyUI_windows_portable>.\python_embeded\python.exe -s ComfyUI\main.py --windows-standalone-build
[START] Security scan
[DONE] Security scan
## ComfyUI-Manager: installing dependencies done.
** ComfyUI startup time: 2024-10-21 02:06:12.666732
** Platform: Windows
** Python version: 3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 10:12:12) [MSC v.1938 64 bit (AMD64)]
** Python executable: C:\ComfyUI_windows_portable\python_embeded\python.exe
** ComfyUI Path: C:\ComfyUI_windows_portable\ComfyUI
** Log path: C:\ComfyUI_windows_portable\comfyui.log

Prestartup times for custom nodes:
   0.0 seconds: C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\rgthree-comfy
   0.0 seconds: C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Easy-Use
   1.5 seconds: C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager

Total VRAM 12288 MB, total RAM 32607 MB
pytorch version: 2.5.0+cu124
WARNING[XFORMERS]: xFormers can't load C++/CUDA extensions. xFormers was built for:
    PyTorch 2.4.1+cu124 with CUDA 1204 (you have 2.5.0+cu124)
    Python  3.11.9 (you have 3.11.9)
  Please reinstall xformers (see https://github.com/facebookresearch/xformers#installing-xformers)
  Memory-efficient attention, SwiGLU, sparse and more won't be available.
  Set XFORMERS_MORE_DETAILS=1 for more details
xformers version: 0.0.28.post1
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3060 : cudaMallocAsync
Using pytorch cross attention`

**Comment by LukeG89** [comment]
Do you really need xformers? As you said, ComfyUI is working good because now it uses Pytorch cross attention.
Just uninstall xformers and keep using Pytorch.

**Reply by chnisar515** [reply]
LukeG89 Thank you again for guide me. i have uninstalled xformers and then again i reinstall it and now its working. but now when i try to launch comfyui it did not load (Launch) automatically on default browser even it showing me only the address to open it on manually on browser. and the 2nd thing is when i try to draw and drop any workflow it wont load in my comfyui even it showing me disabled icon while i am trying to load workflow.

**Reply by LukeG89** [reply]
> i have uninstalled xformers and then again i reinstall it and now its working.

So now it says "xformers cross attention"? Or "pytorch cross attention"?

> but now when i try to launch comfyui it did not load (Launch) automatically on default browser

I don't know what can be the cause.

> and the 2nd thing is when i try to draw and drop any workflow it wont load in my comfyui even it showing me disabled icon while i am trying to load workflow.

Can you screenshot it?

**Reply by chnisar515** [reply]
> So now it says "xformers cross attention"? Or "pytorch cross attention"?
[START] Security scan
[DONE] Security scan
## ComfyUI-Manager: installing dependencies done.
** ComfyUI startup time: 2024-10-21 02:47:18.203344
** Platform: Windows
** Python version: 3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 10:12:12) [MSC v.1938 64 bit (AMD64)]
** Python executable: C:\ComfyUI_windows_portable\python_embeded\python.exe
** ComfyUI Path: C:\ComfyUI_windows_portable\ComfyUI
** Log path: C:\ComfyUI_windows_portable\comfyui.log

Prestartup times for custom nodes:
   0.0 seconds: C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\rgthree-comfy
   0.0 seconds: C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Easy-Use
   1.6 seconds: C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager

Total VRAM 12288 MB, total RAM 32607 MB
pytorch version: 2.5.0+cu124
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3060 : cudaMallocAsync
Using pytorch cross attention
[Prompt Server] web root: C:\ComfyUI_windows_portable\ComfyUI\web
Adding extra search path checkpoints E:\comfyuimodels\models/checkpoints/
Adding extra search path clip E:\comfyuimodels\models/clip/
Adding extra search path clip_interrogator E:\comfyuimodels\models/clip_interrogator/
Adding extra search path clip_vision E:\comfyuimodels\models/clip_vision/
Adding extra search path controlnet E:\comfyuimodels\models/controlnet/
Adding extra search path diffusers E:\comfyuimodels\models/diffusers/
Adding extra search path diffusion_models E:\comfyuimodels\models/diffusion_models/
Adding extra search path embeddings E:\comfyuimodels\models/embeddings/
Adding extra search path gligen E:\comfyuimodels\models/gligen/
Adding extra search path hypernetworks E:\comfyuimodels\models/hypernetworks/
Adding extra search path LLM E:\comfyuimodels\models/LLM/
Adding extra search path llm_gguf E:\comfyuimodels\models/llm_gguf/
Adding extra search path loras E:\comfyuimodels\models/loras/
Adding extra search path animatediff_models E:\comfyuimodels\models/animatediff_models
Adding extra search path photomaker E:\comfyuimodels\models/photomaker/
Adding extra search path vae E:\comfyuimodels\models/vae/
Adding extra search path ipadapter E:\comfyuimodels\models/ipadapter/
C:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\kornia\feature\lightglue.py:44: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  @torch.cuda.amp.custom_fwd(cast_inputs=torch.float32)
Error:
[WinError 1314] A required privilege is not held by the client: 'C:\\ComfyUI_windows_portable\\ComfyUI\\custom_nodes\\ComfyLiterals\\js' -> 'C:\\ComfyUI_windows_portable\\ComfyUI\\web\\extensions\\ComfyLiterals'
Failed to create symlink to C:\ComfyUI_windows_portable\ComfyUI\web\extensions\ComfyLiterals. Please copy the folder manually.
Source: C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyLiterals\js
Target: C:\ComfyUI_windows_portable\ComfyUI\web\extensions\ComfyLiterals
Adding C:\ComfyUI_windows_portable\ComfyUI\custom_nodes to sys.path
Efficiency Nodes: Attempting to add Control Net options to the 'HiRes-Fix Script' Node (comfyui_controlnet_aux add-on)...Success!
Loaded Efficiency nodes from C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\efficiency-nodes-comfyui
Loaded ControlNetPreprocessors nodes from C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui_controlnet_aux
Loaded AdvancedControlNet nodes from C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Advanced-ControlNet
Could not find AnimateDiff nodes
Loaded IPAdapter nodes from C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_IPAdapter_plus
Loaded VideoHelperSuite from C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-VideoHelperSuite
### Loading: ComfyUI-Impact-Pack (V7.10.3)
### Loading: ComfyUI-Impact-Pack (Subpack: V0.7)
### Loading: ComfyUI-Impact-Pack (V7.10.3)
Loaded ImpactPack nodes from[Impact Pack] Wildcards loading done.
C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Impact-Pack
[Impact Pack] Wildcards loading done.
[Crystools INFO] Crystools version: 1.19.0
[Crystools INFO] CPU: 12th Gen Intel(R) Core(TM) i5-12400F - Arch: AMD64 - OS: Windows 10
[Crystools INFO] Pynvml (Nvidia) initialized.
[Crystools INFO] GPU/s:
[Crystools INFO] 0) NVIDIA GeForce RTX 3060
[Crystools INFO] NVIDIA Driver: 565.90
[ComfyUI-Easy-Use] server: v1.2.4 Loaded
[ComfyUI-Easy-Use] web root: C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Easy-Use\web_version/v2 Loaded
Unable to import guidedFilter, make sure you have only opencv-contrib-python or run the import_error_install.bat script
### Loading: ComfyUI-Impact-Pack (V7.10.3)
[Impact Pack] Wildcards loading done.
### Loading: ComfyUI-Inspire-Pack (V1.5.1)
C:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\albumentations\__init__.py:13: UserWarning: A new version of Albumentations is available: 1.4.18 (you have 1.4.15). Upgrade using: pip install -U albumentations. To disable automatic update checks, set the environment variable NO_ALBUMENTATIONS_UPDATE to 1.
  check_for_updates()
C:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\timm\models\layers\__init__.py:48: FutureWarning: Importing from timm.models.layers is deprecated, please import via timm.layers
  warnings.warn(f"Importing from {__name__} is deprecated, please import via timm.layers", FutureWarning)
Total VRAM 12288 MB, total RAM 32607 MB
pytorch version: 2.5.0+cu124
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3060 : cudaMallocAsync
### Loading: ComfyUI-Manager (V2.51.8)
### ComfyUI Revision: 2753 [5f9d5a24] *DETACHED | Released on '2024-10-09'
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/alter-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/model-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/github-stats.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json
--------------
 ### Mixlab Nodes: Loaded
json_repair## OK
ChatGPT.available True
edit_mask.available True
## clip_interrogator_model not found: E:\comfyuimodels\models/clip_interrogator/Salesforce\blip-image-captioning-base, pls download from https://huggingface.co/Salesforce/blip-image-captioning-base
ClipInterrogator.available True
## text_generator_model not found: C:\ComfyUI_windows_portable\ComfyUI\models\prompt_generator\text2image-prompt-generator, pls download from https://huggingface.co/succinctly/text2image-prompt-generator/tree/main
## zh_en_model not found: C:\ComfyUI_windows_portable\ComfyUI\models\prompt_generator\opus-mt-zh-en, pls download from https://huggingface.co/Helsinki-NLP/opus-mt-zh-en/tree/main
PromptGenerate.available True
ChinesePrompt.available True
RembgNode_.available True
TripoSR.available
MiniCPMNode.available
Scenedetect.available
FishSpeech.available
SenseVoice.available
Whisper.available False
fal-client## OK
FalVideo.available
 --------------
Please 'pip install xformers'
Nvidia APEX normalization not installed, using PyTorch LayerNorm
Failed to auto update `Quality of Life Suit`
QualityOfLifeSuit_Omar92_DIR: C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-QualityOfLifeSuit_Omar92
[ReActor] - STATUS - Running v0.5.1-b2 in ComfyUI
Torch version: 2.5.0+cu124
no module 'xformers'. Processing without...
no module 'xformers'. Processing without...
3D OpenPose Editor: Loaded
------------------------------------------
Comfyroll Studio v1.76 :  175 Nodes Loaded
------------------------------------------
** For changes, please see patch notes at https://github.com/Suzie1/ComfyUI_Comfyroll_CustomNodes/blob/main/Patch_Notes.md
** For help, please see the wiki at https://github.com/Suzie1/ComfyUI_Comfyroll_CustomNodes/wiki
------------------------------------------
### [START] ComfyUI AlekPet Nodes v1.0.30 ###

**Reply by chnisar515** [reply]
> Can you screenshot it?
![Screenshot-2024-10-21-031840](https://github.com/user-attachments/assets/c34efca6-9b89-4d79-9962-91215aca0723)

**Reply by LukeG89** [reply]
It seems you have uninstalled xformers.

**Reply by LukeG89** [reply]
About the drag and drop issue, probably there is a conflict with another custom node. If it's a recent problem, I would try disabling the latest nodes you installed. You need to investigate.

**Reply by chnisar515** [reply]
how i can install xformers because i need it in some of my workflow.

**Reply by chnisar515** [reply]
when i try to install Xformers with the this command:

`python_embeded\python.exe -m pip install xformers`

then i got some errors like:

`check_cuda_version
          raise RuntimeError(CUDA_MISMATCH_MESSAGE.format(cuda_str_version, torch.version.cuda))
      RuntimeError:
      The detected CUDA version (11.8) mismatches the version that was used to compile
      PyTorch (12.4). Please make sure to use the same CUDA versions.`

**Reply by LukeG89** [reply]
> how i can install xformers because i need it in some of my workflow.

You need to downgrade pytorch to 2.4.1, because currently there are no compatible xformers for 2.5.0.

This command should both downgrade pytorch and install xformers:
```
python_embeded\python.exe -m pip install torch==2.4.1 torchvision torchaudio xformers --force-reinstall --extra-index-url https://download.pytorch.org/whl/cu124
```

And after this installation, you will probably need to downgrade numpy (I just tried, it auto install a >2 version that is incompatible with ultralytics)
```
python_embeded\python.exe -m pip install numpy<2.0.0 --force-reinstall
```

**Reply by chnisar515** [reply]
This is the latest output when i try to launch ComfyUI

> Total VRAM 12288 MB, total RAM 32607 MB
> pytorch version: 2.5.0+cu124
> WARNING[XFORMERS]: xFormers can't load C++/CUDA extensions. xFormers was built for:
>     PyTorch 2.4.1+cu124 with CUDA 1204 (you have 2.5.0+cu124)
>     Python  3.11.9 (you have 3.11.9)
>   Please reinstall xformers (see https://github.com/facebookresearch/xformers#installing-xformers)
>   Memory-efficient attention, SwiGLU, sparse and more won't be available.
>   Set XFORMERS_MORE_DETAILS=1 for more details
> xformers version: 0.0.28.post1
> Set vram state to: NORMAL_VRAM
> Device: cuda:0 NVIDIA GeForce RTX 3060 : cudaMallocAsync

so please guide me how i can install correct version of Xformers.

**Reply by LukeG89** [reply]
@chnisar515 Follow the comment above

**Reply by chnisar515** [reply]
when i try to use this command

> python_embeded\python.exe -m pip install numpy<2.0.0 --force-reinstall

then i got this output error

> The system cannot find the file specified.

**Reply by LukeG89** [reply]
Are you opening the terminal inside your `ComfyUI_windows_portable` folder?

**Reply by chnisar515** [reply]
yes Sir

**Reply by LukeG89** [reply]
It worked the other command? (downgrading pytorch and installing xformers?)

**Reply by chnisar515** [reply]
after using this command successfully:

> python_embeded\python.exe -m pip install torch==2.4.1 torchvision torchaudio xformers --force-reinstall --extra-index-url https://download.pytorch.org/whl/cu124

i got this output:

> ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
albucore 0.0.16 requires opencv-python-headless>=4.9.0.80, but you have opencv-python-headless 4.7.0.72 which is incompatible.
albumentations 1.4.15 requires opencv-python-headless>=4.9.0.80, but you have opencv-python-headless 4.7.0.72 which is incompatible.
flet 0.24.1 requires packaging<24.0,>=23.1, but you have packaging 24.1 which is incompatible.
gradio 5.1.0 requires markupsafe~=2.0, but you have markupsafe 3.0.2 which is incompatible.
gradio 5.1.0 requires pillow<11.0,>=8.0, but you have pillow 11.0.0 which is incompatible.
gradio 5.1.0 requires typer<1.0,>=0.12; sys_platform != "emscripten", but you have typer 0.9.0 which is incompatible.
inference-cli 0.23.0 requires numpy<=1.26.4, but you have numpy 2.1.2 which is incompatible.
inference-cli 0.23.0 requires rich<=13.5.2, but you have rich 13.9.2 which is incompatible.
inference-gpu 0.23.0 requires numpy<=1.26.4, but you have numpy 2.1.2 which is incompatible.
inference-gpu 0.23.0 requires pillow<11.0, but you have pillow 11.0.0 which is incompatible.
inference-gpu 0.23.0 requires rich<=13.5.2, but you have rich 13.9.2 which is incompatible.
inference-gpu 0.23.0 requires setuptools<=72.1.0,>=70.0.0, but you have setuptools 65.0.0 which is incompatible.
mediapipe 0.10.14 requires protobuf<5,>=4.25.3, but you have protobuf 5.28.2 which is incompatible.
numba 0.60.0 requires numpy<2.1,>=1.22, but you have numpy 2.1.2 which is incompatible.
torchscale 0.3.0 requires fairscale==0.4.0, but you have fairscale 0.4.13 which is incompatible.
torchscale 0.3.0 requires timm==0.6.13, but you have timm 1.0.11 which is incompatible.
Successfully installed MarkupSafe-3.0.2 filelock-3.16.1 fsspec-2024.9.0 jinja2-3.1.4 mpmath-1.3.0 networkx-3.4.1 numpy-2.1.2 pillow-11.0.0 sympy-1.13.3 torch-2.4.1+cu124 torchaudio-2.4.1+cu124 torchvision-0.19.1+cu124 typing-extensions-4.12.2 xformers-0.0.28.post1

and after this when i try to use this command:

> python_embeded\python.exe -m pip install numpy<2.0.0 --force-reinstall

i got this output:

> C:\ComfyUI_windows_portable>python_embeded\python.exe -m pip install numpy<2.0.0 --force-reinstall
The system cannot find the file specified.

C:\ComfyUI_windows_portable>

**Reply by LukeG89** [reply]
The numpy command is not needed. Everything ok when running ComfyUI?

**Reply by chnisar515** [reply]
> The numpy command is not needed. Everything ok when running ComfyUI?

yes now everything working good but many of my nodes are suddenly showing me (IMPORT FAILED).

**Reply by chnisar515** [reply]
> Total VRAM 12288 MB, total RAM 32607 MB
> pytorch version: 2.4.1+cu124
> 
> A module that was compiled using NumPy 1.x cannot be run in
> NumPy 2.1.2 as it may crash. To support both 1.x and 2.x
> versions of NumPy, modules must be compiled with NumPy 2.0.
> Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.
> 
> If you are a user of the module, the easiest solution will be to
> downgrade to 'numpy<2' or try to upgrade the affected module.
> We expect that some modules will need time to support NumPy 2.

**Reply by LukeG89** [reply]
> yes now everything working good but many of my nodes are suddenly showing me (IMPORT FAILED)

Probably because of all the dependency conflicts. Try using the Manager to fix each failed node, or reinstall them.

**Reply by LukeG89** [reply]
> A module that was compiled using NumPy 1.x cannot be run in
> NumPy 2.1.2 as it may crash. To support both 1.x and 2.x
> versions of NumPy, modules must be compiled with NumPy 2.0.
> Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.
> 
> If you are a user of the module, the easiest solution will be to
> downgrade to 'numpy<2' or try to upgrade the affected module.
> We expect that some modules will need time to support NumPy 2.

Here you go... The numpy command is needed 😆
The command should work, I don't know why it says "The system cannot find the file specified." to you.

Try:
`python_embeded\python.exe -m pip uninstall numpy`

and then:
`python_embeded\python.exe -m pip install numpy<2`

**Reply by chnisar515** [reply]
when i use this command, it successfully work to uninstall numpy:

> python_embeded\python.exe -m pip uninstall numpy

and here is the output:

> Found existing installation: numpy 2.1.2
Uninstalling numpy-2.1.2:
  Would remove:
    c:\comfyui_windows_portable\python_embeded\lib\site-packages\numpy-2.1.2-cp311-cp311-win_amd64.whl
    c:\comfyui_windows_portable\python_embeded\lib\site-packages\numpy-2.1.2.dist-info\*
    c:\comfyui_windows_portable\python_embeded\lib\site-packages\numpy.libs\libscipy_openblas64_-c16e4918366c6bc1f1cd71e28ca36fc0.dll
    c:\comfyui_windows_portable\python_embeded\lib\site-packages\numpy.libs\msvcp140-23ebcc0b37c8e3d074511f362feac48b.dll
    c:\comfyui_windows_portable\python_embeded\lib\site-packages\numpy\*
    c:\comfyui_windows_portable\python_embeded\scripts\f2py.exe
    c:\comfyui_windows_portable\python_embeded\scripts\numpy-config.exe
Proceed (Y/n)? y
  Successfully uninstalled numpy-2.1.2
WARNING: Failed to remove contents in a temporary directory 'C:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\~-mpy.libs'.
You can safely remove it manually.
WARNING: Failed to remove contents in a temporary directory 'C:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\~-mpy'.
You can safely remove it manually.

after that when i use this command it did not work:

> C:\ComfyUI_windows_portable>python_embeded\python.exe -m pip install numpy<2
The system cannot find the file specified.

C:\ComfyUI_windows_portable>

**Reply by chnisar515** [reply]
i used this command and its works perfectly and now everything working.

> C:\ComfyUI_windows_portable\python_embeded\python.exe -m pip install "numpy<2"

**Reply by LukeG89** [reply]
@chnisar515 Perfect!
